beamer/agent/chain.py

```diff
@@ -99,7 +105,15 @@ def _thread_func(self) -> None:
         self._call_on_sync_done()
         self._log.info("Sync done")
         while not self._stop:
-            events = fetcher.fetch()
+            try:
+                events = fetcher.fetch()
+                if self._latest_rpc_status is False:
+                    self._latest_rpc_status = True
+                    self._call_on_rpc_status_change(True)
+            except (ConnectionError, HTTPError):
```

Comment:
Another question: why do we catch `HTTPError`?

Reply:
I saw HTTPError on EventFetcher is just there in _fetch_range raising in a scenario but it is not handled so would just crash the agent so i just added this also here

Reply:
> I saw HTTPError on EventFetcher is just there in _fetch_range raising in a scenario but it is not handled so would just crash the agent so i just added this also here

I would actually prefer in this case to crash the agent and know what happened, i.e. the RPC might still be working, but it could be that they don't like some of our requests. If we just ignore those, we will never find out about such RPCs and will therefore not be able to help the LPs using those RPCs. For that reason, I would catch only `ConnectionError` here.

Reply:
ok